org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/alloy4/OurAntiAlias.java

```diff
@@ -92,25 +98,43 @@ public void paint(Graphics gr) {
 	 * Constructs an antialias-capable JTextPane with a DefaultHighlighter
 	 * associated with it.
 	 * 
-	 * @param attributes - see {@link edu.mit.csail.sdg.alloy4.OurUtil#make
+	 * @param attributes
+	 *            - see {@link edu.mit.csail.sdg.alloy4.OurUtil#make
 	 *            OurUtil.make(component, attributes...)}
 	 */
-	public static JTextPane pane(Object... attributes) {
-		JTextPane ans = new JTextPane() {
-			static final long serialVersionUID = 0;
 
-			@Override
-			public void paint(Graphics gr) {
-				if (antiAlias && gr instanceof Graphics2D) {
-					((Graphics2D) gr).setRenderingHint(RenderingHints.KEY_ANTIALIASING,
-							RenderingHints.VALUE_ANTIALIAS_ON);
+	public static JTextPane pane(Function<MouseEvent, String> tooltip, Object... attributes) {
+		try {
+			JTextPane ans = new JTextPane() {
+				static final long serialVersionUID = 0;
+
+				@Override
+				public void paint(Graphics gr) {
+					if (antiAlias && gr instanceof Graphics2D) {
+						((Graphics2D) gr).setRenderingHint(RenderingHints.KEY_ANTIALIASING,
+								RenderingHints.VALUE_ANTIALIAS_ON);
+					}
+					super.paint(gr);
 				}
-				super.paint(gr);
-			}
-		};
-		OurUtil.make(ans, attributes);
-		ans.setHighlighter(new DefaultHighlighter());
-		map.put(ans, Boolean.TRUE);
-		return ans;
+
+				@Override
+				public String getToolTipText(MouseEvent event) {
+					if (tooltip != null) {
+						return tooltip.apply(event);
+					}
+					return super.getToolTipText(event);
+				}
+
+			};
+			if (tooltip != null)
+				ToolTipManager.sharedInstance().registerComponent(ans);
+
+			OurUtil.make(ans, attributes);
+			ans.setHighlighter(new DefaultHighlighter());
+			map.put(ans, Boolean.TRUE);
+			return ans;
+		} catch (Exception e) {
```

Comment:
How can an exception possibly happen here?  Catching and swallowing base `Exception` seems dangerous, and can easily hide simple programming errors (like NPE).

Reply:
In this case the parser can throw an exception I do not care about. I just want to be sure the tooltip does not destroy the GUI in any possible ways. We need to add SLF4J so we can log these things.